config/config.go

```diff
@@ -0,0 +1,145 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/config/eigendaflags"
+	eigendaflags_v2 "github.com/Layr-Labs/eigenda-proxy/config/eigendaflags/v2"
+	"github.com/Layr-Labs/eigenda-proxy/metrics"
+	"github.com/Layr-Labs/eigenda-proxy/server"
+	"github.com/Layr-Labs/eigenda-proxy/store"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore"
+	"github.com/Layr-Labs/eigenda-proxy/verify/v1"
+	"github.com/urfave/cli/v2"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	clients_v2 "github.com/Layr-Labs/eigenda/api/clients/v2"
+)
+
+// AppConfig ... Highest order config. Stores
+// all relevant fields necessary for running both proxy
+// & metrics servers.
+type AppConfig struct {
+	EigenDAConfig ProxyConfig
+	MetricsCfg    metrics.Config
+}
+
+func (c AppConfig) Check() error {
+	err := c.EigenDAConfig.Check()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadCLIConfig(ctx *cli.Context) AppConfig {
+	return AppConfig{
+		EigenDAConfig: ReadProxyConfig(ctx),
+		MetricsCfg:    metrics.ReadConfig(ctx),
+	}
+}
+
+// ProxyConfig ... Higher order config which bundles all configs for instrumenting
+// the proxy server with necessary client context
+type ProxyConfig struct {
+	ServerConfig        server.Config
+	EdaV1ClientConfig   clients.EigenDAClientConfig
+	EdaV1VerifierConfig verify.Config
+
+	EdaV2ClientConfig common.V2ClientConfig
+
+	MemstoreConfig memstore.Config
+	StorageConfig  store.Config
+	PutRetries     uint
+
+	MemstoreEnabled bool
+
+	EigenDAV2Enabled bool
```

Comment:
add comments saying that eigendav2 can only be enabled when memstore is not enabled.

Reply:
technically not true with the current implementation - as of now memstore can only be enabled for V1

Reply:
Added comment to capture: [20944e3](https://github.com/Layr-Labs/eigenda-proxy/pull/296/commits/20944e32884209950be9ff3c4ef017945ca3fc94)